flink-java/src/main/java/org/apache/flink/api/java/utils/Option.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.api.java.utils;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Internal representation of a parameter passed to a user defined function.
+ */
+public class Option {
+
+	private String longName;
+	private String shortName;
+
+	private String defaultValue;
+	private Set<String> choices;
+
+	private String helpText;
+	private OptionType type = OptionType.STRING;
+
+	public Option(String name) {
+		this.longName = name;
+		this.choices = new HashSet<>();
+	}
+
+	/**
+	 * Define a alternative / short name of the Parameter.
+	 *
+	 * @param shortName - short version of the parameter name
+	 * @return the updated Option
+	 */
+	public Option alt(String shortName) {
```

Comment:
I don't see a reason why more than one alternative should be helpful. I'll add a comment to point out that only one alternative is allowed.